### DISCUSSION TITLE
Log-in to Edit?

### DISCUSSION THREAD

**Discussion by SonicDH** [cu/silicon#12] [Ideas]
I know you're using this personally, so in that case having a log-in system is silly. But I'm using this as a knowledge base for the staff of a small convention and I don't really want just anyone to be able to edit it however they want. Is there any way to implement a login system just for editing?

**Comment by cu** [comment]
It might be possible, I'll have to look into it. Are you using anything for authentication now? (If so, what?)

**Reply by SonicDH** [reply]
Right now it's running dockerized in Unraid on my private network, so no inherent authentication other than the fact people just can't access it for the moment.

**Comment by cu** [comment]
Silicon doesn't currently have any notion of authentication or authorization. But I think it is possible to kinda fake it without any changes to Silicon itself, however.

I don't know anything about Unraid but over in the [deploy](https://github.com/cu/silicon/tree/master/deploy) directory, there is an example docker-compose deployment called `caddy-authelia`. This is basically how I run Silicon myself. I believe Authelia can be configured to allow anonymous read access to certain URLs/methods and require login for other URLs/methods. See the README there and [these docs](https://www.authelia.com/overview/authorization/access-control/) for an overview. Since there are only two URLs that modify content in silicon, I believe it would be possible to configure an access control policy that allowed access to all Silicon URLs, but require login for those two. They are:

* `^/edit/.*$` (GET, POST)
* `^/related/.*$` (POST, DELETE)

I would imagine it would not be difficult to configure something similar in nginx or other web servers, perhaps just using HTTP Basic Authentication if you want to keep it simple. I know this answer might be more involved than you were hoping for but it's the best I have right now. :)
